### PR TITLE
Basic dependency fetching

### PR DESCRIPTION
**Pull request by sewe2000**
I implemented basic dependency fetching. I can say that it works for the test case scenario I created (my emacs-wayland rule depends on Nasz-Elektryk/nerd-fonts rule). On the other hand it doesn't fetch rules asynchronously and the code may require solid improvement. Also we should discuss how asynchronous fetch can be further achieved. Additionally I fixed some annoying bugs along the way.

### CODE REVIEW

File: src/checker/ruleset_location.go [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+	if error != nil {
+		return []string{}, error
+	}
+
+	lockfile.Write(yamlOutput)
+
+	return outputDependencyTree.Dependencies, nil
+}
+
+func anyToError(val any) error {
```
Comment (BaderBC) [inline src/checker/ruleset_location.go:171]: Apart from the fact that it seems weird to create new utils file only for this function, I do not think `anyToError` should placed in this file especially that it isn't even used in `ruleset_location.go`

File: src/checker/check.go [hunk 2]
```diff
@@ -102,6 +105,15 @@ func _internalCheckRule(importLoopData *ImportLoopData, identifier string, name
 		panic(nil)
 	}
 
+	lockfilePath := ruleSetLocation.getRuleSetPath() + "/" + LOCK_FILENAME
+	if !api.FileExists(lockfilePath, false) {
```
Comment (BaderBC) [inline src/checker/check.go:109]: I will discuss today in the meeting that we shouldn't call `api.*` directly

File: src/checker/ruleset_location.go [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+}
+
+func (rulesetLocation *RuleSetLocation) createLockfile() ([]string, error) {
+	configPath := rulesetLocation.getRuleSetPath() + "/" + CONFIG_FILENAME
+	configFileContents, error := os.ReadFile(configPath)
+	if error != nil {
+		return []string{}, error
+	}
+
+	type DependencyTreeLayout struct {
```
Comment (BaderBC) [inline src/checker/ruleset_location.go:104]: I believe it should be also accessible for other function

File: src/checker/ruleset_location.go [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+		dependencyLocation := RuleSetLocation{}
+		dependencyLocation.new(strings.Split(dependencyName, "@")[0])
+		if !dependencyLocation.isRuleSetDownloaded() {
+			error = FetchRuleSet(&dependencyLocation)
+			if error != nil {
+				return []string{}, error
+			}
+		}
+		lockfilePath := dependencyLocation.getRuleSetPath() + "/" + LOCK_FILENAME
+		if api.FileExists(lockfilePath, false) {
```
Comment (BaderBC) [inline src/checker/ruleset_location.go:128]: As I said, avoid `api.*`

**Review by BaderBC — CHANGES_REQUESTED** [review]
Looks good to me currently, waiting for asynchronous fetching based on lock-file

**Review by cooligus — APPROVED** [review]

**Pull request merged**